Fabric.Terminology.Domain/Models/ValueSet.cs

```diff
@@ -2,23 +2,103 @@
 {
     using System.Collections.Generic;
 
+    using Fabric.Terminology.Domain.Services;
+
     public class ValueSet : IValueSet
     {
-        public string ValueSetUniqueId { get; set; }
+        public ValueSet(
+            string valueSetId,
```

Comment:
Yep - already has created problems actually and one of the reasons I went to a fluent pattern in the Client project (making the constructors internal).  I'll add this as a to do in the future.